### PR TITLE
Separate native-only feature for `wgpu::CommandEncoder::write_timestamp`

### PR DESCRIPTION
**Pull request by Wumpf**
**Connections**
* https://github.com/gpuweb/gpuweb/pull/4370

**Description**
WebGPU no longer supports timestamps on encoder (the only kind supported now are the ones on render/compute passes), meaning we have to split up out `TIMESTAMP_QUERY_INSIDE_ENCODERS` of the existing `TIMESTAMP_QUERY` feature.

Note that this gets our feature number at 64, meaning I had to switch the feature flag datatype to u128 (.. this will solve the problem _once and for all_! ;))

**Testing**

There's a test on timestamp that I ran on Mac. Change overall fairly straight forward, low risk.

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown`
  - [ ] `--target wasm32-unknown-emscripten`
- [x] Run `cargo xtask test` to run tests.
- [x] Add change to `CHANGELOG.md`. See simple instructions inside file.

### CODE REVIEW

**Comment by ErichDonGubler** [comment]
I'll take this one.

**Comment by Wumpf** [comment]
@ErichDonGubler how's that progressing? :)

File: examples/src/mipmap/mod.rs [hunk 1]
```diff
@@ -8,8 +8,8 @@ const MIP_PASS_COUNT: u32 = MIP_LEVEL_COUNT - 1;
 
 const QUERY_FEATURES: wgpu::Features = {
     wgpu::Features::TIMESTAMP_QUERY
-        .union(wgpu::Features::PIPELINE_STATISTICS_QUERY)
         .union(wgpu::Features::TIMESTAMP_QUERY_INSIDE_PASSES)
+        .union(wgpu::Features::PIPELINE_STATISTICS_QUERY)
```
Comment (ErichDonGubler) [inline examples/src/mipmap/mod.rs:12]: question: 🤔 Why did we swap the order here and in docs.? Is this to, say, appease a dependency on `TIMESTAMP_QUERY` before requesting `TIMESTAMP_QUERY_INSIDE_ENCODERS`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -89,7 +89,9 @@ Bottom level categories:
 - Eager release of GPU resources comes from device.trackers. By @bradwerth in [#5075](https://github.com/gfx-rs/wgpu/pull/5075)
 - `wgpu-types`'s `trace` and `replay` features have been replaced by the `serde` feature. By @KirmesBude in [#5149](https://github.com/gfx-rs/wgpu/pull/5149)
 - `wgpu-core`'s `serial-pass` feature has been removed. Use `serde` instead. By @KirmesBude in [#5149](https://github.com/gfx-rs/wgpu/pull/5149)
-
+- `wgpu::CommandEncoder::write_timestamp` requires now the new `wgpu::Features::TIMESTAMP_QUERY_INSIDE_ENCODERS` feature which is available
+  on all native backends but not on WebGPU (due to a spec change `write_timestamp` is no longer supported on WebGPU).
+  By @wumpf in [#](??)
```
Comment (ErichDonGubler) [inline CHANGELOG.md]: style: This should all be on one line, to match the rest of the document.

**Review by ErichDonGubler — APPROVED** [review]
Approving, conditional on `style` feedback being addressed and resolution of CI issues. I'm not concerned about the rest of the PR. Oh, except:

- [x] Can we please file follow-up work to investigate switching `InstanceFlags` to `enumset`, per [discussion in Matrix](https://matrix.to/#/!FZyQrssSlHEZqrYcOb:matrix.org/$fVc-TvibKjMJT1mqkgS7iMH-cN8UzU0-ITIotZ8lHQY?via=matrix.org&via=mozilla.org&via=envs.net) noting that `u128` can be problem for `web*` backends?

**Comment by ErichDonGubler** [comment]
> @ErichDonGubler how's that progressing? :)

I'm so sorry for taking so long. I've been caught up in a shorter work week this past week, and wrangling some of my other contributions to this repo. 😅 Thanks for the ping and holding me accountable!

File: examples/src/mipmap/mod.rs (hunk 1 above)
Comment (Wumpf) [reply examples/src/mipmap/mod.rs:12]: yeah just cosmetics essentially. When I went through all the usages of timestamp this made it easier to get an overview

**Comment by Wumpf** [comment]
oops, messed up a feature check. But luckily the test prevent that from going in, phew

**Pull request merged**